core/src/main/java/org/jdbi/v3/core/locator/ClasspathSqlLocator.java

```diff
@@ -75,7 +79,11 @@ private static String resourcePathFor(String fullyQualifiedName) {
     }
 
     private static String findSqlOnClasspath(ClassLoader classLoader, String path) {
-        return CACHE.computeIfAbsent(path, p -> readResource(classLoader, p));
+        try {
+            return CACHE.get(path, () -> readResource(classLoader, path));
```

Comment:
Can't do that -- the `classLoader` is not available there, and it's not part of the cache key, so we have to capture it.